WordPress/gutenberg
# Post featured image: Add title attribute

## What?

This PR adds an option to add a title attribute to the post featured image. The setting is in the Advanced panel in the block settings sidebar.

## Why?

There are inconsistencies between the available features for the image block and the post featured image block.
This PR is meant to reduce one of these inconsistencies to make it easier for users to use the blocks.

## How?

- Adds a new block attribute called title.
- Adds a new text input control in the Advanced panel.

## Testing Instructions
1. Create a new post and assign it a featured image.
2. Add a post featured image block. -The "Link to post" setting should be toggled off.
3. With the block selected, open the Advanced panel in the block settings sidebar.
4. Locate the TITLE ATTRIBUTE setting and add a text. Save.
5. Refresh the editor, open the advanced panel, and confirm that the attribute has saved correctly.
6. View the post on the front of the website, confirm that the image has a title attribute, for example by viewing the source.
7. Go back to the editor and enable the "Link to post setting". Save and view the post on the front:
- If the post has a title, the linked image should not have a title attribute. This is to ensure that the post title in the alt attribute is used as the link text / accessible name.

## Screenshots or screencast 
![The block settings sidebar for the post featured image block with the new title attribute setting in the Advanced panel.](https://user-images.githubusercontent.com/7422055/210545173-cd88e2d1-5de5-4e25-9dbf-081e9aed39a9.png)
